File: tournaments/isOneSwapEnough/isOneSwapEnough.cs
bool isOneSwapEnough(string inputString) {
    if (inputString.Length==1)
        return true;
    char[]a=inputString.ToCharArray();
    for (int i=0; i<a.Length;i++)
        for (int j=0; j<a.Length;j++){
            char tmp=a[i];
            a[i]=a[j];
            a[j]=tmp;
            if (isP(a, a.Length))return true;
            tmp=a[i];
            a[i]=a[j];
            a[j]=tmp;
        }
    return false;
}

bool isP(char[]a, int n){
    for (int i=0; i<n; i++)
        if (a[i]!=a[n-i-1])return false;
    return true;
}

File: tournaments/isOneSwapEnough/isOneSwapEnough.js
function isOneSwapEnough(inputString) {
    const isP = function(a) {
        const n = a.length;

        for (let i = 0; n > i; ++i)
            if (a[i] !== a[n - i - 1]) return false;

        return true;
    };

    const a = [...inputString];
    const n = a.length;

    if (1 === n)
        return true;

    for (let i = 0; n > i; ++i)
        for (let j = 0; n > j; ++j) {
            [a[i], a[j]] = [a[j], a[i]];
            if (isP(a)) return true;
            [a[i], a[j]] = [a[j], a[i]];
        }

    return false;
}
